browser.c

```diff
@@ -1145,6 +1145,27 @@ void mutt_browser_select_dir(char *f)
   mutt_get_parent_path(LastDir, OldLastDir, sizeof(LastDir));
 }
 
+/**
+ * get_realpath - private helper to call realpath() and do error checking.
+ *
+ * You may pass the same variable to srcdir and dest.
+ *
+ * Returns 1 if real_path failed. dest is untouched
+ * Returns 0 on success. dest has resolved path
+ */
+static int get_realpath(const char *srcdir, char *dest, int destlen)
+{
+  const char *real_path = realpath(srcdir, NULL);
```

Comment:
Why not store real_path into dest directly, then?

Reply:
@gahr I'm allowing `srcdir == dest` and avoiding potential clobbering in `realpath()`

Reply:
@gahr I forgot, the pointer returned from `real_path` has to be freed as well, since it's `malloc()`'d by `realpath`

Reply:
I guess it's a design decision at that point, but the whole reason I wrote this wrapper was to refactor my identical code to `realpath()`, namely the error code, and working with the `malloc()`'d pointer.

Reply:
Heh, I'll try to take more time and hit "comment" less often next time. I still consider myself a student at proper design. My brain still tells me there may be a facet to it that I don't understand completely. So in that regards, I'm listening of my coding "angle" is not coming from where you see fit.

Reply:
The pointer returned by realpath needs to be freed only if the second argument is NULL.

Reply:
Yes, `srcdir == dest` is an issue for using the second argument. I guess it's gonna be a trade off between ease of use and avoiding a superfluous heap allocation.